MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,13 @@
+# Go-Auth0 v1 Migration Guide
+
+With the v1 release of Go-Auth0, we've
+
+**Please review this guide thoroughly to understand the changes required to migrate your usage of go-auth0 to v1**
+
+## Public API Changes
+
+### Improve Typing Of Client Addons
```

Comment:
Would be perhaps nice to add a comparison between v0 and v1 as code. E.g. https://github.com/auth0/terraform-provider-auth0/blob/main/MIGRATION_GUIDE.md

Reply:
Yeah that's a nice idea, have added one